packages/examples/packages/name-lookup/src/index.ts

```diff
@@ -0,0 +1,19 @@
+import type { OnNameLookupHandler } from '@metamask/snaps-types';
+
+export const onNameLookup: OnNameLookupHandler = async (request) => {
+  const { chainId, address, domain } = request;
+
+  if (address) {
+    const shortAddress = `0x${address.substring(2, 5)}`;
+    const chainIdHex = `0x${parseInt(chainId.split(':')[1], 10).toString(16)}`;
+    const resolvedDomain = `Test Token - ${shortAddress} / ${chainIdHex}`;
```

Comment:
This isn't a domain?

Reply:
It technically supports any string so we're using it in the extension to propose any kind of name for addresses.

I've updated this example to `test.domain` though for clarity.

Reply:
I would have preferred this to be just the domain string returned, maybe I need to update the SIP to specify this. The idea would be to return exactly what the registry of the protocol you're querying has the record indexed by. I'm not sure if I feel comfortable with adding some opinion about proposed names onto the API layer. That's something the client (us) should handle on the layer that wants to use the `resolvedDomain` in some manner, which is petnames in our case.

Reply:
I couldn't find anything about it in the spec, so please update it!